src/views/Dashboard.jsx

```diff
@@ -0,0 +1,309 @@
+import { Fragment, useState } from "react";
+import { Link, useNavigate, Outlet } from "react-router-dom";
+
+import { Dialog, Menu, Transition } from "@headlessui/react";
+import {
+  Bars3Icon,
+  HomeIcon,
+  XMarkIcon,
+  PlusCircleIcon,
+} from "@heroicons/react/24/outline";
+import { ChevronDownIcon } from "@heroicons/react/20/solid";
+import { auth } from "../../firebase_config";
+import { signOut } from "firebase/auth";
+
+function classNames(...classes) {
+  return classes.filter(Boolean).join(" ");
+}
+
+function Dashboard() {
+  const [sidebarOpen, setSidebarOpen] = useState(false);
+  const [currentTab, setCurrentTab] = useState("");
+
+  const currentUser = auth.currentUser;
+  console.log(currentUser);
+
+  const navigate = useNavigate();
+
+  const navigation = [
+    {
+      name: "all posts",
+      href: "all-posts",
+      icon: HomeIcon,
+      current: currentTab === "all posts",
+    },
+    {
+      name: "new post",
+      href: "create",
+      icon: PlusCircleIcon,
+      current: currentTab === "new post",
+    },
+  ];
+
+  const onLogout = async (e) => {
```

Comment:
here the `async` is declared but doesn't use the `await` keyword inside. If you're not using any asynchronous operations inside, you can remove the `async` keyword or change it to be asynchronous.